pkg/util/cluster/aad.go

```diff
@@ -6,57 +6,44 @@ package cluster
 import (
 	"context"
 	"fmt"
-	"net/http"
 	"time"
 
-	azgraphrbac "github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
-	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/date"
+	msgraph_apps "github.com/microsoftgraph/msgraph-sdk-go/applications"
+	msgraph_models "github.com/microsoftgraph/msgraph-sdk-go/models"
+	msgraph_errors "github.com/microsoftgraph/msgraph-sdk-go/models/odataerrors"
 	"k8s.io/apimachinery/pkg/util/wait"
-
-	"github.com/Azure/ARO-RP/pkg/util/uuid"
 )
 
-func (c *Cluster) getServicePrincipal(ctx context.Context, appID string) (string, error) {
-	// TODO: we are listing here rather than calling
-	// i.applications.GetServicePrincipalsIDByAppID() due to some missing
-	// permission with our dev/e2e applications
-	sps, err := c.serviceprincipals.List(ctx, fmt.Sprintf("appId eq '%s'", appID))
+func (c *Cluster) createApplication(ctx context.Context, displayName string) (string, string, error) {
```

Comment:
Rhetorical side question - what's the point of setting a display name?  

I mean we've always been using it and we can trace it back to ARO cluster applications.

Reply:
It looks nice in the Azure portal?  Actually we could be embedding more information into it nowadays like OpenShift version or API visibility.